library/Reporting/Web/Forms/ReportForm.php

```diff
@@ -100,6 +120,28 @@ protected function assemble()
         }
     }
 
+    public function validate()
+    {
+        parent::validate();
+
+        if (! $this->isValid) {
+            return false;
+        }
+
+        $select = (new Select())
+            ->from('report r')
+            ->columns(new Expression('1'))
+            ->where(['r.name = ?' => $this->getValue('name')]);
+
+        if ($this->getDb()->select($select)->fetch() !== false) {
+            $this->addMessage($this->translate(sprintf('Report "%s" already exists', $this->getValue('name'))));
+
+            $this->isValid = false;
+        }
```

Comment:
Unfortunately, this prevents not only from having duplicate names, but also from any report updates. So, if you want to edit an existing report, e.g. because you want to have a different breakdown, it will fail.

In order to fix this, I would suggest that the `ReportForm` is not only passed report id, but the whole report instance. See how this works in the `ScheduleForm` and do it the same way here. Then you can validate the names in the `validate()` as follows.
```php
$name = $this->getValue('name');
if (! $this->report || $this->report->getName() !== $name) {
     $select = ...
}
```